dist-tools/create-all-services.js

```diff
@@ -0,0 +1,161 @@
+var fs = require('fs');
+var path = require('path');
+
+var defaultServices = 'acm,apigateway,applicationautoscaling,autoscaling,cloudformation,cloudfront,cloudhsm,cloudtrail,cloudwatch,cloudwatchlogs,cloudwatchevents,codecommit,codedeploy,codepipeline,cognitoidentity,cognitoidentityserviceprovider,cognitosync,configservice,devicefarm,directconnect,dynamodb,dynamodbstreams,ec2,ecr,ecs,elasticache,elasticbeanstalk,elastictranscoder,elb,emr,firehose,gamelift,inspector,iotdata,kinesis,kms,lambda,marketplacecommerceanalytics,mobileanalytics,machinelearning,opsworks,rds,redshift,route53,route53domains,s3,ses,sns,sqs,ssm,storagegateway,sts,waf';
+var sanitizeRegex = /[^a-zA-Z0-9,-]/;
+
+// Generate service clients
+function ClientCreator() {
+  this._metadata = require('../apis/metadata');
+  this._apisFolderPath = path.join(__dirname, '..', 'apis');
+  this._clientFolderPath = path.join(__dirname, '..', 'browser');
+  this._serviceCustomizationsFolderPath = path.join(__dirname, '..', 'lib', 'services');
+  this._apiFileNames = null;
+}
+
+ClientCreator.prototype.getAllApiFilenames = function getAllApiFilenames() {
+    if (this._apiFileNames) {
+        return this._apiFileNames;
+    }
+    var apiFileNames = fs.readdirSync(this._apisFolderPath);
+    // filter out metadata
+    this._apiFileNames = apiFileNames.filter(function(name) {
```

Comment:
Just a minor comment, but it never made much sense to me to have `metadata.json` in the `apis/` directory where service models were. It was always the oddball in that directory, and always required extra logic (like this bit of code here) to exclude it from being counted as a service. It makes sense to me to move `metadata.json` to the root of `lib/`, and would allow us to get rid of this extra filter.